performance_test/src/data_running/data_runner_base.hpp

```diff
@@ -39,7 +44,28 @@ class DataRunnerBase : boost::noncopyable
 public:
   DataRunnerBase()
   : m_ec(ExperimentConfiguration::get())
-  {}
+  {
+    if (m_ec.check_memory()) {
+#ifdef MEMORY_TOOLS_ENABLED
+      osrf_testing_tools_cpp::memory_tools::initialize();
+
+      auto on_unexpected_memory =
+              [](osrf_testing_tools_cpp::memory_tools::MemoryToolsService &service) {
+                std::cerr << "unexpected malloc";
+                // this will cause a backtrace to be printed for each unexpected memory operations
+                service.print_backtrace();
+              };
+      osrf_testing_tools_cpp::memory_tools::on_unexpected_calloc(on_unexpected_memory);
+      osrf_testing_tools_cpp::memory_tools::on_unexpected_free(on_unexpected_memory);
+      osrf_testing_tools_cpp::memory_tools::on_unexpected_malloc(on_unexpected_memory);
+      osrf_testing_tools_cpp::memory_tools::on_unexpected_realloc(on_unexpected_memory);
+
```

Comment:
`on_unexpected_realloc`

Sorry for the comments' placements, it's probably my Chrome.

Reply:
Np.